packager/live_packager.cc

```diff
@@ -379,6 +380,12 @@ Status LivePackager::Package(const Segment& init_segment,
   packaging_params.cts_offset_adjustment =
       config_.format == LiveConfig::OutputFormat::TS;
 
+  if (config_.decryption_keys) {
```

Comment:
While it's convenient to use this pointer to indicate source decryption, I am not a fan of the raw pointer / memory management.  Who's responsible for freeing the memory?

Perhaps consider declaring this as a regular object and later check that `key_map` is not empty.